core/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -467,6 +480,50 @@ public InetAddress nextUp(Object value) {
             public InetAddress nextDown(Object value) {
                 return InetAddressPoint.nextDown((InetAddress)value);
             }
+
+            @Override
+            public BytesRef encodeRanges(Set<Range> ranges) throws IOException {
+                final byte[] encoded = new byte[ByteUtils.MAX_BYTES_VLONG + (16 * 2) * ranges.size()];
+                ByteArrayDataOutput out = new ByteArrayDataOutput(encoded);
+                out.writeVInt(ranges.size());
```

Comment:
since we use a fixed length, maybe we do not need to encode the number of values?